src/replication.c

```diff
@@ -2634,6 +3387,11 @@ void syncWithPrimary(connection *conn) {
             if (err) goto write_error;
         }
 
+        /* Announce that the replica is capable of dual channel sync. */
+        if (server.dual_channel_replication) {
+            err = sendCommand(conn, "REPLCONF", "capa", "dual-channel", NULL);
+        }
```

Comment:
This err is never checked.

Missing `if (err) goto write_error;`?

Reply:
Btw, this doesn't matter, but I believe it should be possible to send this together with all the other capas. The primary will ignore capabilities it does not understand.

Reply:
Right, I'll add a check